Fix implicit fallthrough warning on GCC 7.x

afirago opened the pull request:
Add fallthrough comments to keep GCC >= 7.x happy.

Signed-off-by: Alexey Firago <alexey_firago@mentor.com>

grpc-testing commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-testing commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-jenkins commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-testing commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-testing commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-jenkins commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-testing commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-testing commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-jenkins commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-jenkins commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

afirago commented:
CLA is updated

grpc-testing commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-testing commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

grpc-testing commented:
Thanks for your pull request. The automated tests will run as soon as one of the admins verifies this change is ok for us to run on our infrastructure.

ctiller commented on src/core/lib/support/murmur_hash.c:
```diff
@@ -61,15 +61,15 @@ uint32_t gpr_murmur_hash3(const void *key, size_t len, uint32_t seed) {
+      k1 ^= ((uint32_t)tail[2]) << 16; //-fallthrough
+    case 2: //-fallthrough
+      k1 ^= ((uint32_t)tail[1]) << 8; //-fallthrough
+    case 1: //-fallthrough
```
surely this doesn't need to happen on every line of the case stanza?

afirago replied:
@ctiller thanks for checking. Fixed it to required minimum of comments.

ctiller replied:
Would it read better as:
```c
    case 2:
       k1 ^= ((uint32_t)tail[1]) << 8; 
       //-fallthrough
```

afirago replied:
@ctiller done!

ctiller commented:
Jenkins: this is ok to test

grpc-jenkins commented:
```
[trickle] No significant performance differences
```

grpc-jenkins commented:
```
[microbenchmarks] No significant performance differences
```

grpc-jenkins commented:
```
[trickle] No significant performance differences
```

ctiller commented:
Care to make one last change to keep our clang-format checks consistent:
```
   switch (len & 3) {
     case 3:
       k1 ^= ((uint32_t)tail[2]) << 16;
-      //-fallthrough
+    //-fallthrough
     case 2:
       k1 ^= ((uint32_t)tail[1]) << 8;
-      //-fallthrough
+    //-fallthrough
     case 1:
       k1 ^= tail[0];
       k1 *= c1;
```

ctiller commented:
(I think clang-format is wrong here, but it's better to keep it happy)

afirago commented:
@ctiller  
Should I also change "//-fallthrough" comments to /* fallthrough */ to be consistent with the coding style?

Also I've found several similar issues in other parts of the source tree. Should I add more commits to this PR or close this one and open new with one commit covering all "fallthrough" warnings?

ctiller commented:
/* fallthrough */ would be better for us, although I'm not terribly fussy.
Feel free to append this PR with more fixes - I can test them all in a
batch.

We have a wrapper script, tools/distrib/clang_format_code.sh that does all
the code formatting in a consistent way (it'll need docker to be installed)
if that's helpful for getting it all polished.

On Thu, Jul 13, 2017 at 2:18 PM Alexey Firago <notifications@github.com>
wrote:

> @ctiller <https://github.com/ctiller>
> Should I also change "//-fallthrough" comments to /* fallthrough */ to be
> consistent with the coding style?
>
> Also I've found several similar issues in other parts of the source tree.
> Should I add more commits to this PR or close this one and open new with
> one commit covering all "fallthrough" warnings?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/grpc/grpc/pull/11796#issuecomment-315205085>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AJpudezxS8tiwLeQ84uWk1m0oFq2iXSSks5sNommgaJpZM4OXLQ3>
> .
>

afirago commented:
Updates:
* Fixed indentation as requested
* Changed "//-fallthrough" to preferred "/* fallthrough */ 
* Add more fixes to the commit for the same issue across the source tree. Updated commit and PR name

grpc-jenkins commented:
```
[trickle] No significant performance differences
```

grpc-jenkins commented:
```
[microbenchmarks] No significant performance differences
```
